core/src/pixelcolor/conversion.rs

```diff
@@ -1,8 +1,18 @@
 use crate::pixelcolor::{binary_color::*, gray_color::*, rgb_color::*};
 
 /// Convert color channel values from one bit depth to another.
-const fn convert_channel(value: u8, from_max: u8, to_max: u8) -> u8 {
-    ((value as u16 * to_max as u16 + from_max as u16 / 2) / from_max as u16) as u8
+///
+/// Fixed point implementation of the conversion formula:
+/// out = round(in * from_max / to_max)
```

Comment:
Minor readability improvement?:

```suggestion
/// `out = round(in * from_max / to_max)`
```